### PR TITLE
Version 1

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

**Review by KostiantynVorobiov — APPROVED** [review]
As for me OK

File: src/main/java/com/internet/shop/Main.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    public static void main(String[] args) {
+        ProductService productService = (ProductService) injector.getInstance(ProductService.class);
+        productService.create(new Product("PS4", 500));
+        productService.create(new Product("XBOX", 450));
+        productService.create(new Product("Nintendo", 300));
+        System.out.println("All Products:");
+        Storage.products.forEach(System.out::println);
+        System.out.println();
+        productService.deleteById(2L);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/Main.java]: save your products to variables and use `someProduct.getId()` instead everywhere

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                .findFirst();
+    }
+
+    @Override
+    public Product update(Product product) {
+        for (int i = 0; i < Storage.products.size(); i++) {
+            if (Storage.products.get(i).getId().equals(product.getId())) {
+                return Storage.products.set(i, product);
+            }
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: the logic is ok, but let's use Stream API

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return null;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: return `product` in each case

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public boolean deleteById(Long productId) {
+        for (int i = 0; i < Storage.products.size(); i++) {
+            if (Storage.products.get(i).getId().equals(productId)) {
+                Storage.products.remove(i);
+            }
+        }
+        return true;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: we can use `removeIf()` on storage

File: src/main/java/com/internet/shop/model/Order.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.List;
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public Order(Long id) {
+        this.id = id;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Order.java]: the id is set in Storage class
same for other models

File: src/main/java/com/internet/shop/model/Product.java [hunk 6]
```diff
@@ -1,9 +1,9 @@
-package shop.model;
+package com.internet.shop.model;
 
 import java.math.BigDecimal;
 
 public class Product {
-    private long id;
+    private Long id;
     private String name;
     private BigDecimal price;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Product.java]: let's have `double` here for now

File: src/main/java/com/internet/shop/dao/ProductDao.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.internet.shop.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+public interface ProductDao {
+
+    Product create(Product product);
+
+    Optional<Product> geById(Long productId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/ProductDao.java]:
```suggestion
    Optional<Product> getById(Long productId);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
remove classes with solution from `master` branch

and here, in this PR, let's remove `<dependencies>` block from `pom.xml`
we do not use dependencies yet

File: src/main/java/com/internet/shop/Main.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/Main.java:19]: like this
Product playStation =  productService.create(new Product("PS4", 500));

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: done

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: ok

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: done

File: src/main/java/com/internet/shop/model/Order.java (hunk 5 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/model/Order.java]: done. changed also in Cart

File: src/main/java/com/internet/shop/model/Product.java (hunk 6 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/model/Product.java]: done

File: src/main/java/com/internet/shop/dao/ProductDao.java (hunk 7 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/dao/ProductDao.java]: done

File: src/main/java/com/internet/shop/Main.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/Main.java]: done.

File: src/main/java/com/internet/shop/Main.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        Product playStation = new Product("PS4", 500);
+        productService.create(playStation);
+        productService.create(xbox);
+        productService.create(nintendo);
+
+        System.out.println("All Products:");
+        Storage.products.forEach(System.out::println);
+        System.out.println("getById 1");
+        System.out.println(playStation.getId());
+        productService.deleteById(2L);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/Main.java]:
```suggestion
        productService.deleteById(nintendo.getId());
```

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        return getAllProducts().stream()
+                .filter(product -> product.getId().equals(productId))
+                .findFirst();
+    }
+
+    @Override
+    public Product update(Product product) {
+        return getAllProducts().stream()
+                .filter(p -> p.getId().equals(product.getId()))
+                .findFirst().get();
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: where are you updating?

let's create IntStream of indices, than filter out only those indices on which there are elements with the same id
then for each of the remaining indices, replace the element with our `product`

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public boolean deleteById(Long productId) {
+        for (int i = 0; i < Storage.products.size(); i++) {
+            if (Storage.products.get(i).getId().equals(productId)) {
+                Storage.products.removeIf(product -> product.getId().equals(productId));
+            }
+        }
+        return true;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: too much redundant code
`removeIf()` already iterates through all elements and returns boolean

File: src/main/java/com/internet/shop/model/Order.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.List;
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public Order(Long userId) {
+        this.userId = userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Order.java:12]: let's initialize `products` in the constructor

File: src/main/java/com/internet/shop/model/ShoppingCart.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.List;
+
+public class ShoppingCart {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public ShoppingCart(Long userId) {
+        this.userId = userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/ShoppingCart.java:12]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**